viewer/packages/tools/src/Measurement/MeasurementControls.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Cognite3DViewer } from '@reveal/core';
+import * as THREE from 'three';
+import { MeasurementGizmo } from './MeasurementGizmo';
+import { MeasurementLabel } from './MeasurementLabel';
+import { Measurement } from './Measurement';
+import { MeasurementLineOptions } from './types';
+import { MeasurementDistance } from './MeasurementDistance';
+
+export class MeasurementControls {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _measurementGizmo: MeasurementGizmo;
+  private readonly _measurementLabel: MeasurementLabel;
+  private readonly _startPosition: THREE.Vector3;
+  private _measurement: Measurement;
+  private _pointSize: number;
+
+  private readonly _handleonPointerClick = this.onPointerClick.bind(this);
+  private readonly _handleonPointerMove = this.onPointerMove.bind(this);
+
+  constructor(viewer: Cognite3DViewer) {
+    this._viewer = viewer;
+    this._measurementGizmo = new MeasurementGizmo(this._viewer);
+    this._measurementLabel = new MeasurementLabel(this._viewer);
+    this._startPosition = new THREE.Vector3();
+    this._pointSize = 0.02;
+  }
+
+  /**
+   * Set input handling
+   */
+  private setupEventHandling() {
+    this._viewer.on('click', this._handleonPointerClick);
+  }
+
+  /**
+   * Remove input handling
+   */
+  private removeEventHandling() {
+    this._viewer.off('click', this._handleonPointerClick);
+  }
+
+  /**
+   * Add an measurement to the control
+   * @param measurement Measurement object which is the active measurement type
+   */
+  add(measurement: Measurement): void {
+    if (this._measurement) {
+      this.remove();
+    }
+    this._measurement = measurement;
+    this.setupEventHandling();
+  }
+
+  /**
+   * Remove measurement from control
+   */
+  remove(): void {
+    if (this._measurement) {
+      this._measurement.remove();
+      this._measurement = null;
+      this.removeEventHandling();
+    }
+  }
+
+  private async onPointerClick(event: MouseEvent) {
+    const { offsetX, offsetY } = event;
+    const pointer = new THREE.Vector2(offsetX, offsetY);
+
+    const intersection = await this._viewer.getIntersectionFromPixel(pointer.x, pointer.y);
```

Comment:
Why create a vector2 from something you just access .x and .y from the line after?